From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Wed, 23 Sep 2026 16:16:02 -0400
Subject: [PATCH] python: Use "free" from ctypes.pythonapi instead of searching
 for libc

This fixes an incompatibility with Windows/MSVC, where libc is not
available using `find_library("c")` but we can access the one function
needed with pythonapi.

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>
---
 bindings/python/iio/__init__.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/bindings/python/iio/__init__.py b/bindings/python/iio/__init__.py
index fc06a58e..2a0f4099 100644
--- a/bindings/python/iio/__init__.py
+++ b/bindings/python/iio/__init__.py
@@ -367,7 +367,6 @@ else:
 if _lib_loc is None:
     raise OSError(2, "Could not find libiio C library")
 _lib = _cdll(_lib_loc, use_errno=True, use_last_error=True)
-_libc = _cdll(find_library("c"))
 
 _get_backends_count = _lib.iio_get_builtin_backends_count
 _get_backends_count.restype = c_uint
@@ -817,7 +816,7 @@ _ev_get_channel.restype = _ChannelPtr
 _ev_get_channel.argtypes = (_EventPtr, _DevicePtr, c_bool)
 _ev_get_channel.errcheck = _check_null
 
-_libc_free = _libc.free
+_libc_free = ctypes.pythonapi.free
 _libc_free.argtypes = (c_void_p,)
 
 # pylint: enable=invalid-name
-- 
2.51.0

